src/core/components/files-regular.js

```diff
@@ -232,21 +232,24 @@ module.exports = function (self) {
 
     pull(
       exporter(ipfsPath, self._ipld, options),
+      pull.filter(filterFile),
+      pull.take(1),
       pull.collect((err, files) => {
-        if (err) { return d.abort(err) }
-        if (files && files.length > 1) {
-          files = files.filter(filterFile)
+        if (err) {
+          return d.abort(err)
         }
-        if (!files || !files.length) {
+
+        if (!files.length) {
           return d.abort(new Error('No such file'))
         }
 
         const file = files[0]
-        const content = file.content
-        if (!content && file.type === 'dir') {
+
+        if (!file.content && file.type === 'dir') {
```

Comment:
These conditions might be simpler as:

```suggestion
        if (file.type !== 'file') {
          return d.abort(new Error('this dag node was not a file')
        }
        
        if (!file.content) {
          return d.abort(new Error('this dag node had no content')
        }
```

Though they could also be added to the `filterFile` function.

Reply:
Fair enough.  I guess once all the errors have codes we can stop depending on message content.

If `file.type === 'file'` then `file.content` should always have a value.  It's weird that there is a check for that in the first place 🤷‍♂️.